coretrace/interaction.cpp

```diff
@@ -133,202 +133,236 @@ void Interaction(
 /*{  InteractionType = 1, Refraction
 ===============================================================================}*/
 	case 1:
-			Refr1 = Opticl->RefractiveIndex[0];
-			Refr2 = Opticl->RefractiveIndex[2];
-			RMU = Refr1/Refr2;
-			D2 = DOT(DFXYZ,DFXYZ);
-			B = (RMU*RMU - 1.0)/D2;
-			A = RMU*DOT(CosKLM, DFXYZ) /D2;
-			A2 = A*A;
-			if (B > A2)     //Total internal reflection
-			{
-				A = DOT(CosKLM, DFXYZ)/DOT(DFXYZ, DFXYZ);
-				for (i=0; i<3; i++)
-					CosOut[i] = CosKLM[i] - 2.0*A*DFXYZ[i];
-				return;
-			}
+	{
+		Refr1 = Opticl->RefractiveIndex[0];
```

Comment:
We need to create a test that uses refraction. The parabolic trough receiver glass envelope uses this feature. I can put it on the short list to add.